lib/improver/nbhood/square_kernel.py

```diff
@@ -119,7 +119,7 @@ def cumulate_array(cube):
         xname = cube.coord(axis="x").name()
         cubelist = iris.cube.CubeList([])
         nan_masks = []
-        data = cube.data
```

Comment:
It might be worth running a test using the parameters we use in suite (or running one topographic nbhood suite task) to see if you hit memory issues if you haven't done so already.

Reply:
I haven't tried this. Would it be possible for you to test it.

Reply:
I ran the suite for one cycle with enukx and ukvx and I didn't encounter any memory issues.